source/grid.js

```diff
@@ -18,22 +18,24 @@
 
 this.initialize = function() {
     //Set up the grid map as a 2D array
-    this.tiles = [];
-    for ( var i = 0; i < this.maxX - this.minX; i++ ) {
-        this.tiles[ i ] = [];
-        for ( var j = 0; j < this.maxY - this.minY; j++ ) {
-            this.tiles[ i ][ j ] = new GridTile();
+    var tiles = this.tiles = new Array( this.maxX - this.minX );
+    for ( var i = 0; i < tiles.length; i++ ) {
+        tiles[ i ] = new Array( this.maxY - this.minY );
+        for ( var j = 0; j < tiles[ i ].length; j++ ) {
+            tiles[ i ][ j ] = new GridTile();
         }
     }
     this.future( 0 ).setBoundaryValues();
 }
 
 this.setBoundaryValues = function() {
-    if ( this.boundaryValues ) {
+    var boundaryValues = this.boundaryValues;
+    var tiles = this.tiles;
```

Comment:
Since `setBoundaryValues` is a property, each reference is a trip through the kernel. Keeping a local reference lets the iteration work with just one property access.

The mapping that makes node references work happens on every property reference. Since node mapping now walks through the entire property, it's important to avoid redundant access to properties that contain large objects.

You could avoid the explicit local reference for `setBoundaryValues` (but not `tiles`) by iterating like this:

    this.setBoundaryValues = function() {
        var tiles = this.tiles;
        ( this.boundaryValues || [] ).forEach( function( boundaryValuesX, i ) {
            boundaryValuesX && boundaryValuesX.forEach( function( boundaryValuesXY, j ) {
                tiles[ i ][ j ].energyRequired = boundaryValuesXY;
            }, this );
        }, this );